contracts/SimpleSettlement.sol

```diff
@@ -3,42 +3,173 @@
 
 /**
  * @title Simple Settlement contract
  * @notice Contract to execute limit orders settlement, created by Fusion mode.
  */
-contract SimpleSettlement is ResolverValidationExtension, IntegratorFeeExtension {
+contract SimpleSettlement is FeeTaker {
+    uint256 private constant _BASE_POINTS = 10_000_000; // 100%
+    uint256 private constant _GAS_PRICE_BASE = 1_000_000; // 1000 means 1 Gwei
+
+    error AllowedTimeViolation();
+
     /**
      * @notice Initializes the contract.
      * @param limitOrderProtocol The limit order protocol contract.
-     * @param feeToken The token to charge protocol fees in.
      * @param accessToken Contract address whose tokens allow filling limit orders with a fee for resolvers that are outside the whitelist.
      * @param weth The WETH address.
      * @param owner The owner of the contract.
      */
-    constructor(address limitOrderProtocol, IERC20 feeToken, IERC20 accessToken, address weth, address owner)
-        BaseExtension(limitOrderProtocol)
-        ResolverValidationExtension(feeToken, accessToken, owner)
-        IntegratorFeeExtension(weth)
-        Ownable(owner)
+    constructor(address limitOrderProtocol, IERC20 accessToken, address weth, address owner)
+        FeeTaker(limitOrderProtocol, accessToken, weth, owner)
     {}
 
-    function _postInteraction(
+    /**
+     * @dev Adds dutch auction capabilities to the getter
+     */
+    function _getMakingAmount(
         IOrderMixin.Order calldata order,
         bytes calldata extension,
         bytes32 orderHash,
         address taker,
-        uint256 makingAmount,
         uint256 takingAmount,
         uint256 remainingMakingAmount,
         bytes calldata extraData
-    ) internal virtual override(ResolverValidationExtension, IntegratorFeeExtension) {
-        super._postInteraction(order, extension, orderHash, taker, makingAmount, takingAmount, remainingMakingAmount, extraData);
+    ) internal view override returns (uint256) {
+        (uint256 rateBump, bytes calldata tail) = _getRateBump(extraData);
+        return Math.mulDiv(
+            super._getMakingAmount(order, extension, orderHash, taker, takingAmount, remainingMakingAmount, tail),
+            _BASE_POINTS,
+            _BASE_POINTS + rateBump
+        );
+    }
+
+    /**
+     * @dev Adds dutch auction capabilities to the getter
+     */
+    function _getTakingAmount(
+        IOrderMixin.Order calldata order,
+        bytes calldata extension,
+        bytes32 orderHash,
+        address taker,
+        uint256 makingAmount,
+        uint256 remainingMakingAmount,
+        bytes calldata extraData
+    ) internal view override returns (uint256) {
+        (uint256 rateBump, bytes calldata tail) = _getRateBump(extraData);
+        return Math.mulDiv(
+            super._getTakingAmount(order, extension, orderHash, taker, makingAmount, remainingMakingAmount, tail),
```

Comment:
Looks like we should get fee from the result takingAmount, not intermediate calculations - at first auction calculations, then getting fee calculation